app/src/main/java/com/github/kulebin/myfoursquareapp/useCase/ShowDetailVenueUseCase.java

```diff
@@ -0,0 +1,47 @@
+package com.github.kulebin.myfoursquareapp.useCase;
+
+import android.util.Log;
+
+import com.github.kulebin.myfoursquareapp.api.Api;
+import com.github.kulebin.myfoursquareapp.dataSource.IDataSource;
+import com.github.kulebin.myfoursquareapp.dataSource.IOnResultCallback;
+import com.github.kulebin.myfoursquareapp.model.Venue;
+import com.github.kulebin.myfoursquareapp.presenter.VenueDetailPresentation;
+import com.github.kulebin.myfoursquareapp.view.VenueDisplayData;
+
+public class ShowDetailVenueUseCase {
+
+    private final IDataSource mDataSource;
+    private final VenueDetailPresentation mPresenter;
+
+    public ShowDetailVenueUseCase(final IDataSource pDataSource, final VenueDetailPresentation pPresenter) {
+        mDataSource = pDataSource;
+        mPresenter = pPresenter;
+    }
+
+    public void showDetailVenue() {
+
+        mDataSource.fetchData(Api.getVenuesByIdUrl(mPresenter.getVenueId()), new IOnResultCallback<Venue>() {
```

Comment:
this use case is ok, please update other one. 
Note that here should be no usage of Api class